### PR TITLE
Conditional large initialize

### PR DESCRIPTION
**Pull request by KieranRatcliffeInvertedAI**
Reopend this PR because the previous PR was merged into "master" by accident and was reverted.

### CODE REVIEW

**Comment by KieranRatcliffeInvertedAI** [comment]
I've added one more feature to this PR for infraction support. If it looks good, let me know so that I can merge this PR into develop.

File: invertedai/large/initialize.py [hunk 1]
```diff
@@ -309,45 +309,76 @@ def _insert_agents_into_nearest_regions(
...
             if nearby_region is not None:
                 if sqrt((nearby_region.center.x-region.center.x)**2+(nearby_region.center.y-region.center.y)**2) > (REGION_MAX_SIZE + AGENT_SCOPE_FOV_BUFFER):
                     continue
             region_agent_states = region.agent_states
-            agent_states = agent_states + [state for i, state in enumerate(region_agent_states) if i not in region_agent_remove_map[ind]]
-            agent_properties = agent_properties + [prop for i, prop in enumerate(region.agent_properties[:len(region_agent_states)]) if i not in region_agent_remove_map[ind]]
-            recurrent_states = recurrent_states + [recurr for i, recurr in enumerate(region.recurrent_states) if i not in region_agent_remove_map[ind]]
-    
-    return agent_states, agent_properties, recurrent_states
+            agent_states = agent_states + [state for state in region_agent_states]
```
Comment (Ruishenl) [inline invertedai/large/initialize.py:378]: why not just `agent_states + region_agent_states` ?

Comment (KieranRatcliffeInvertedAI) [reply invertedai/large/initialize.py:378]: Good catch, this was a leftover from splitting an earlier function into 2. I'll make this change.

File: invertedai/large/initialize.py [hunk 2]
```diff
@@ -274,29 +240,147 @@ def get_number_of_agents_per_region_by_drivable_area(
...
+    for prop in agent_properties[num_agent_states:]:
+        random_region_index = randint(0,num_regions-1)
+        regions[random_region_index].agent_properties.append(prop)
+
+        if return_region_index: region_map.append(tuple([random_region_index,len(regions[random_region_index].agent_properties)-1]))
+
+    return regions, region_map
+
+def _consolidate_all_responses(
+    all_responses: Optional[InitializeResponse] = None,
```
Comment (rf-ivtdai) [inline invertedai/large/initialize.py:313]: I see all_responses is assumed to be a list below with `len` and indexing, but it's of type `InitializeResponse`, am I tripping here?

Comment (KieranRatcliffeInvertedAI) [reply invertedai/large/initialize.py:313]: Huh that is weird that it didn't cause an issue during testing...
Anyways it is fixed now

**Review by Ruishenl — APPROVED** [review]

**Pull request merged**